front/src/components/Task/TaskBtnGroup.vue

```diff
@@ -1,48 +1,96 @@
 <template>
-  <div class="q-gutter-sm">
-    <q-btn
-      v-touch-hold:2000.mouse="handleOnItClick"
-      round
-      size="sm"
-      :title="onItTitle"
-      :icon="onItIcon"
-      :color="onItColor"
-      @click="updateOnIt(!onIt)"
-    />
-    <q-btn
-      round
-      size="sm"
-      title="Enter flag"
-      icon="flag"
-      color="positive"
-      @click="solveTask(task)"
-    />
-    <q-btn
-      round
-      size="sm"
-      :title="`Edit ${task.title}`"
-      icon="edit"
-      color="warning"
-      @click="editTask(task)"
-    />
-    <q-btn
-      round
-      size="sm"
-      :title="`Delete ${task.title}`"
-      icon="delete"
-      color="negative"
-      @click="deleteTask(task)"
-    />
+  <div>
+    <div v-if="!dense" class="q-gutter-sm">
+      <q-btn
+        v-touch-hold:2000.mouse="handleOnItClick"
+        round
+        size="sm"
+        :title="onItTitle"
+        :icon="onItIcon"
+        :color="onItColor"
+        @click="updateOnIt(!onIt)"
+      />
+      <q-btn
+        round
+        size="sm"
+        title="Enter flag"
+        icon="flag"
+        color="positive"
+        @click="solveTask(task)"
+      />
+      <q-btn
+        round
+        size="sm"
+        :title="`Edit ${task.title}`"
+        icon="edit"
+        color="warning"
+        @click="editTask(task)"
+      />
+      <q-btn
+        round
+        size="sm"
+        :title="`Delete ${task.title}`"
+        icon="delete"
+        color="negative"
+        @click="deleteTask(task)"
+      />
+    </div>
+
+    <q-btn-group v-if="dense">
+      <q-btn
+        :to="openTaskRoute"
+        round
+        size="sm"
+        :title="`Open ${task.title}`"
+        icon="edit_note"
+        color="primary"
+      />
+      <q-btn
+        v-touch-hold:2000.mouse="handleOnItClick"
+        round
+        size="sm"
+        :title="onItTitle"
+        :icon="onItIcon"
+        :color="onItColor"
+        @click="updateOnIt(!onIt)"
+      />
+      <q-btn
+        round
+        size="sm"
+        title="Enter flag"
+        icon="flag"
+        color="positive"
+        @click="solveTask(task)"
+      />
+      <q-btn
+        round
+        size="sm"
+        :title="`Edit ${task.title}`"
+        icon="edit"
+        color="warning"
+        @click="editTask(task)"
+      />
+      <q-btn
+        round
+        size="sm"
+        :title="`Delete ${task.title}`"
+        icon="delete"
+        color="negative"
+        @click="deleteTask(task)"
+      />
```

Comment:
These buttons are exactly the same as the buttons when it is not dense. Can you put them into a template a refer to them again at both places so you don't have duplicated code?